src/components/GoodsList.js

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+
+const goodsFromServer = [
+  'Dumplings',
+  'Carrot',
+  'Eggs',
+  'Ice cream',
+  'Apple',
+  'Bread',
+  'Fish',
+  'Honey',
+  'Jam',
+  'Garlic',
+];
+
+class GoodsList extends React.Component {
+  state = {
+    listGoods: [],
+    isLoaded: false,
+  };
+
+  handleClick = () => {
+    this.setState({
+      listGoods: [...goodsFromServer],
+      isLoaded: true,
+    });
+  };
+
+  getReverse = () => {
+    this.setState({
+      listGoods: [...goodsFromServer].reverse(),
+    });
+  };
+
+  getSort = () => {
+    this.setState({
+      listGoods: [...goodsFromServer].sort(
+        (a, b) => a.localeCompare(b)
+      ),
+    });
+  };
+
+  getReset = () => {
+    this.setState({
+      listGoods: [...goodsFromServer],
+    });
+  };
+
+  getSortByLength = () => {
+    this.setState({
+      listGoods: [...goodsFromServer].sort(
+        (a, b) => a.length - b.length
+      ),
+    });
+  };
+
+  setGoodsLength = (event) => {
+    this.setState({
+      listGoods: [...goodsFromServer].filter(
```

Comment:
filter и так создает копию массива
```suggestion
      listGoods: goodsFromServer.filter(
```